jwtBearer.go

```diff
@@ -39,7 +39,7 @@ type jwtBearer struct {
 	errMutex *sync.RWMutex
 }
 
-func NewClientWithJWTBearer(sandbox bool, instanceURL, consumerKey, username string, privateKey []byte, tokenDuration time.Duration, httpClient http.Client) (Client, error) {
+func NewClientWithJWTBearer(sandbox bool, instanceURL, consumerKey, username string, privateKey []byte, tokenDuration time.Duration, httpClient http.Client) (*jwtBearer, error) {
```

Comment:
I changed this return type from an interface type (`Client`) to a struct type (`*jwtBearer`) to comply with [Go client libraries](https://bookstack.niche.team/books/back-end-patterns-practices/page/inter-service-communication#bkmrk-go-client-libraries).

I didn't think it was particularly valuable to also make sure that the returned struct type was exported, as I'm not sure what the use would be for other modules to be able to reference the jwtBearer _type_ directly. Let me know if you think I'm missing something and should convert this to an exported struct type!

Reply:
It seems reasonable to me! I agree that ohter modules are unlikely to have a need to reference the jwtBearer type directly.